fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/QueryToKeyMatcher.java

```diff
@@ -375,8 +374,6 @@ private Match matches(@Nonnull OneOfThemWithComparison query, @Nonnull KeyExpres
             } else {
                 return matches(query, children.get(0), matchingMode, filterMask);
             }
-        } else if (key instanceof GroupingKeyExpression) {
```

Comment:
This is not possible. `GroupingKeyExpression` can only occur at the top and is properly dealt with by the first overload.